web-ng/src/app/components/map/map.component.ts

```diff
@@ -46,9 +47,15 @@ import firebase from 'firebase/app';
 
 const normalIconScale = 30;
 const enlargedIconScale = 50;
+const zoomedInLevel = 13;
 const normalPolygonStrokeWeight = 3;
 const enlargedPolygonStrokeWeight = 6;
-const zoomedInLevel = 13;
+
+type Polygon = google.maps.Polygon & { id: string };
+
+function Polygon(id: string, polygon: google.maps.Polygon): Polygon {
```

Comment:
Can you just use `google.maps.Polygon` directly and call `.set('id', id)` and `get('id')` to retain the id?

Reply:
it does not have a option of set and get id https://developers.google.com/maps/documentation/javascript/reference/polygon#PolylineOptions

Reply:
[google.maps.Polygon](https://developers.google.com/maps/documentation/javascript/reference/polygon#Polygon) which you're using here extends [MVCObject](https://developers.google.com/maps/documentation/javascript/reference/event#MVCObject), which has `get()` and `set()` methods. Why can't these method be used?

Reply:
right implemented was following some other link.

Reply:
Oh I gave the suggestion of extending Polygon with `id` field, didn't realize that MVCObject has `get()` & `set()` for storing the ID.  Sorry for the detour @HemanParbhakar.  This should look much cleaner, thanks @gino-m.